src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -1332,11 +1332,54 @@ def _reassign_generic_particle_states(self, archive, logger) -> None:
                 new.normalize(archive, logger)
                 self.particle_states.append(new)
 
+    def _validate_subsystem(self, logger: 'BoundLogger') -> None:
+        """ """
+
+        if self.is_root_system():
+            return
+
+        if self.particle_indices is None:
+            logger.warning(
+                'Cannot validate ModelSystem subsystem without particle_indices.'
+            )
+            return
+
+        parent = self.m_parent
+        if parent.is_root_system():
+            particle_array = (
+                parent.positions or parent.particle_states or parent.velocities
```

Comment:
is this possible that you have positions or velocities but no particle states? should you just check particle_states?

Reply:
IMO positions should be the main quantity, one could want to store some configuration of positions without labeling them explicitly. Of course our parsers would not do that, and maybe it would break all the normalization anyway so it is moot.

@EBB2675 what do you think?